CONTRIBUTING.md

```diff
@@ -77,6 +77,31 @@ Or run the documentation and browse to [http://localhost:1234/](http://localhost
 yarn start:docs
 ```
 
+### Component/Hook scaffolding
+If you are looking to contribute a brand new component or hook in a package that does not exist yet, please run the following command.
+```bash
+yarn plop
+```
+This will start a series of cli prompts to determine what template files and folders should be generated to help quick start your contribution.
+The prompts are as follows:
+```
+1. What type of project are you setting up?
+- Select "React Spectrum v3"
+
+2. scope name(s)
+- Select the package scopes that make sense for your contribution (e.g. react-spectrum if you are contributing a component, react-aria for an aria hook, etc).
+
+3. Package name, all lowercase (e.g. textfield)
+
+4. Component name, please use appropriate uppercase (e.g. TextField)
+- If you are contributing a non-component hook, just enter the name of your hook.
+
+5. Component css module name, blank if N/A. If unsure, check @adobe/spectrum-css-temp/components for a module containing the desired css (e.g. textfield)
+- If you cannot find the component css module name in @adobe/spectrum-css-temp/components, feel free to reach out to a team member on the git issue.
+```
+
+Upon answering all the prompts, the appropriate package(s) should be generated and ready for modification. Note: With regards to the generated docs files, please feel free to remove them if you won't be contributing docs. If you are contributing docs for a new component/hook, please submit them in a separate PR so that we can decide when to deploy them to our docs site.
```

Comment:
```suggestion
Upon answering all the prompts, the appropriate package(s) should be generated and ready for modification.

**Note:** With regards to the generated docs files, please feel free to remove them if you won't be contributing docs. If you are contributing docs for a new component or Hook, please submit them in a separate pull request so that we can decide when to deploy them to our docs site.
```